tests/infra/ccf.py

```diff
@@ -347,9 +347,9 @@ def create_and_trust_node(
             new_node.stop()
             return None
 
+        new_node.network_state = NodeNetworkState.joined
```

Comment:
This was a small bug in the test infra as we were not waiting for _the_ new node to actually catch up before continuing.